Python/ceval_gil.c

```diff
@@ -527,24 +527,22 @@ PyStatus
 _PyEval_InitGIL(PyThreadState *tstate, int own_gil)
 {
     assert(tstate->interp->ceval.gil == NULL);
-    int locked;
     if (!own_gil) {
         /* The interpreter will share the main interpreter's instead. */
         PyInterpreterState *main_interp = _PyInterpreterState_Main();
         assert(tstate->interp != main_interp);
         struct _gil_runtime_state *gil = main_interp->ceval.gil;
         init_shared_gil(tstate->interp, gil);
-        locked = current_thread_holds_gil(gil, tstate);
+        assert(!current_thread_holds_gil(gil, tstate));
```

Comment:
This change implies that we never call `_PyEval_InitGIL()` with the GIL held.  That seems reasonable.  Do you know why we might have been accommodating the other case?

Reply:
I'm not sure why we were accommodating the other case. It's possible we will want it in the future.